geocoding_android/android/src/main/java/com/baseflow/geocoding/MethodCallHandlerImpl.java

```diff
@@ -168,31 +192,44 @@ private void onPlacemarkFromCoordinates(final MethodCall call, final Result resu
         final double longitude = call.argument("longitude");
 
         try {
-            final List<Address> addresses = geocoding.placemarkFromCoordinates(
+            geocoding.placemarkFromCoordinates(
                     latitude,
-                    longitude);
+                    longitude, new GeocodeListenerAdapter() {
+
+                        @Override
+                        public void onGeocode(List<Address> addresses) {
+                            if (addresses != null && addresses.size() > 0) {
+                                result.success(AddressMapper.toAddressHashMapList(addresses));
+                            } else {
+                                result.error(
+                                        "NOT_FOUND",
+                                        String.format(
+                                                Locale.ENGLISH,
+                                                "No address information found for supplied coordinates (latitude: %f, longitude: %f).",
+                                                latitude,
+                                                longitude
+                                        ),
+                                        null);
+                            }
+                        }
+
+                        @Override
+                        public void onError(String errorMessage) {
+                            result.error(
+                                    "NOT_FOUND",
```

Comment:
Practically these will always be IO errors, so I think it would be better to rename this to `IO_ERROR` to prevent confusion:
```suggestion
                            "IO_ERROR",
```